### PR TITLE
Andy/migration refactor

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

File: go/libraries/doltcore/doltdb/root_val.go [hunk 1]
```diff
@@ -277,9 +277,11 @@ func (root *RootValue) GenerateTagsForNewColumns(ctx context.Context, tableName
 	}
 
 	newTags := make([]uint64, len(newColNames))
+	existingTags := set.NewUint64Set(rootSuperSchema.AllTags())
 	for i := range newTags {
-		newTags[i] = schema.AutoGenerateTag(rootSuperSchema, tableName, existingColKinds, newColNames[i], newColKinds[i])
+		newTags[i] = schema.AutoGenerateTag(existingTags, tableName, existingColKinds, newColNames[i], newColKinds[i])
 		existingColKinds = append(existingColKinds, newColKinds[i])
+		existingTags.Add(newTags[i])
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/doltdb/root_val.go:284]: @bheni this is why `irs-soi` was failing. It had so many columns that it was creating tag collisions while generating tags for a single table. Updating the existing tag set on each tag creation fixes it.

File: bats/compatibility/corona-virus-test.sh [hunk 2]
```diff
@@ -99,7 +99,7 @@ local_bin="`pwd`"/"$bin"
 PATH="$local_bin":"$PATH" dolt clone Liquidata/corona-virus
 pushd "corona-virus"
 PATH="$local_bin":"$PATH" export_tables "-pre"
-dolt migrate
+time dolt migrate
```
Comment (bheni) [inline bats/compatibility/corona-virus-test.sh:102]: did you want this in the checked in file?

File: go/libraries/doltcore/rebase/rebase_tag.go [hunk 3]
```diff
@@ -608,91 +716,83 @@ func validateTagMapping(tagMapping TagMapping) error {
...
 			return false, nil
 		})
 
-		newTags, err := rebasedParentRoot.GenerateTagsForNewColumns(ctx, tn, colNames, colKinds)
-		if err != nil {
-			return err
-		}
-		if len(oldTags) != len(newTags) {
-			return errors.New("error generating unique tags for migration")
+		// generate tags with the same mether as root.GenerateTagsForNewColumns()
```
Comment (bheni) [inline go/libraries/doltcore/rebase/rebase_tag.go:776]: mether?

Comment (andrew-wm-arthur) [reply go/libraries/doltcore/rebase/rebase_tag.go:776]: oof

**Pull request merged**